core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -136,7 +137,10 @@ public class RemoteLogManager implements Closeable {
 
     // topic ids that are received on leadership changes, this map is cleared on stop partitions
     private final ConcurrentMap<TopicPartition, Uuid> topicPartitionIds = new ConcurrentHashMap<>();
+    private final String clusterId;
 
+    // The endpoint for remote log metadata manager to connect to
+    private Optional<EndPoint> endpoint = Optional.empty();
```

Comment:
This shouldn't be optional as discussed in prior comments. Our code logic enforces that this is always present (else we throw an error) at
```
endpoint = brokerInfo.broker.endPoints.find(e => e.listenerName.equals(listenerName))
            .getOrElse(throw new ConfigException(RemoteLogManagerConfig.REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP +
              " should be set as a listener name within valid broker listener name list."))
```

Reply:
Right. Updated.